daly-bms-uart.cpp

```diff
@@ -39,63 +40,277 @@ bool Daly_BMS_UART::Init()
     return true;
 }
 
-bool Daly_BMS_UART::getPackMeasurements(float &voltage, float &current, float &SOC)
+bool Daly_BMS_UART::update()
 {
-    this->sendCommand(COMMAND::VOUT_IOUT_SOC);
+
+    getPackMeasurements(); //0x90
+    getMinMaxCellVoltage(); //0x91
+    getPackTemp();  //0x92
+    getDischargeChargeMosStatus(); //0x93
+    getStatusInfo(); //0x94
+    //getCellVoltages(); //dont work, the answer string from BMS is too long and must splitet
+    getFailureCodes(); //0x98
+    /**
+ * put here the function call to recive all data one by one
+ * check if cell number ar set and then ask for cell data
+ * 
+ */
 
     if (!this->receiveBytes())
     {
 #ifdef DALY_BMS_DEBUG
-        Serial.printf("<DALY-BMS DEBUG> Receive failed, V, I, & SOC values won't be modified!\n");
+        DEBUG_SERIAL.print("Updating the Data Failed\n");
 #endif
         return false;
     }
+    return true;
+}
+
+bool Daly_BMS_UART::getPackMeasurements() //0x90
+{
+    this->sendCommand(COMMAND::VOUT_IOUT_SOC);
 
+    if (!this->receiveBytes())
+    {
+#ifdef DALY_BMS_DEBUG
+        DEBUG_SERIAL.print("<DALY-BMS DEBUG> Receive failed, V, I, & SOC values won't be modified!\n");
+#endif
+        return false;
+    }
     // Pull the relevent values out of the buffer
-    voltage = (float)((this->my_rxBuffer[4] << 8) | this->my_rxBuffer[5]) / 10;
+    get.packVoltage = (float)((this->my_rxBuffer[4] << 8) | this->my_rxBuffer[5]) / 10;
     // The current measurement is given with a 30000 unit offset
-    current = (float)(((this->my_rxBuffer[8] << 8) | this->my_rxBuffer[9]) - 30000) / 10;
-    SOC = (float)((this->my_rxBuffer[10] << 8) | this->my_rxBuffer[11]) / 10;
+    get.packCurrent = (float)(((this->my_rxBuffer[8] << 8) | this->my_rxBuffer[9]) - 30000) / 10;
+    get.packSOC = (float)((this->my_rxBuffer[10] << 8) | this->my_rxBuffer[11]) / 10;
+    return true;
+}
+
+bool Daly_BMS_UART::getMinMaxCellVoltage() //0x91
+{
+    this->sendCommand(COMMAND::MIN_MAX_CELL_VOLTAGE);
+
+    if (!receiveBytes())
+    {
+#ifdef DALY_BMS_DEBUG
+        DEBUG_SERIAL.print("<DALY-BMS DEBUG> Receive failed, min/max cell values won't be modified!\n");
+#endif
+        return false;
+    }
+
+    get.maxCellmV = (float)((this->my_rxBuffer[4] << 8) | this->my_rxBuffer[5]); // Given in mV, convert to V
```

Comment:
```suggestion
    get.maxCellmV = (float)((this->my_rxBuffer[4] << 8) | this->my_rxBuffer[5]);
```

Reply:
This comment is stale because we are no longer dividing by 1000 to convert from mV to V

Reply:
Come to think of it, this value no longer needs to be a float, it could be a uint16_t. The same thing applies for `minCellmV`

Reply:
forgot to remove, sorry